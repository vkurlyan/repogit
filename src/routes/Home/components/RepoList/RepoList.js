import React, {PropTypes} from 'react'
// import './RepoList.scss'
import { ListGroupItem, ListGroup } from 'react-bootstrap';

class RepoList extends React.PureComponent {
    render() {
        return (
            <ListGroup>
                {this.props.repositories.map((repo) => (
                    <ListGroupItem key={String(repo.id)} target="_blank" href={repo.html_url}>
                        <b>{repo.name}</b>
                        {repo.description && `: ${repo.description}`}
                    </ListGroupItem>
                ))}
            </ListGroup>
        )
    }
}

RepoList.propTypes = {
    repositories : PropTypes.array.isRequired,
}

export default RepoList
