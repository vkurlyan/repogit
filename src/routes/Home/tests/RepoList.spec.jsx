import RepoList from '../components/RepoList/RepoList'
import renderer from 'react-test-renderer';
import React from 'react';

const repoListParams = {
    repositories: [
        {
            "id": 1,
            "name": "test-name",
            "html_url": "/1",
            "description": "Description 1",
        },
        {
            "id": 2,
            "name": "test-name-2",
            "html_url": "/2",
        }
    ],
};

describe('RepoList', () => {
    it('shows empty list', () => {
        const component = renderer.create(
            <RepoList repositories={[]} />
        ).toJSON();

        expect(component).toMatchSnapshot();
    });

    it('shows list of repositories', () => {
        const component = renderer.create(
            <RepoList {...repoListParams} />
        ).toJSON();

        expect(component).toMatchSnapshot();
    });
});
