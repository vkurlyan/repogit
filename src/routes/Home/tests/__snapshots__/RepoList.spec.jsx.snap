exports[`RepoList shows empty list 1`] = `
<ul
  className="list-group" />
`;

exports[`RepoList shows list of repositories 1`] = `
<div
  className="list-group">
  <a
    className="list-group-item"
    href="/1"
    target="_blank">
    <b>
      test-name
    </b>
    : Description 1
  </a>
  <a
    className="list-group-item"
    href="/2"
    target="_blank">
    <b>
      test-name-2
    </b>
  </a>
</div>
`;
